service/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java

```diff
@@ -1,63 +1,68 @@
 package com.epam.esm.service.impl;
 
 import com.epam.esm.dao.TagDao;
+import com.epam.esm.dao.UserDao;
+import com.epam.esm.service.exception.CreateEntityInternalException;
 import com.epam.esm.domain.dto.TagDTO;
 import com.epam.esm.domain.entity.Tag;
 import com.epam.esm.service.TagService;
 import com.epam.esm.service.exception.IdNotExistException;
-import com.epam.esm.service.mapper.TagDTOMapper;
+import org.modelmapper.ModelMapper;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
-import java.util.Optional;
 import java.util.stream.Collectors;
 
 @Service
+@Transactional
 public class TagServiceImpl implements TagService {
 
     private final TagDao tagDao;
-    private final TagDTOMapper tagDTOMapper;
+    private final UserDao userDao;

```

Comment:
Service should use only dao that works with its direct entity (Tag here). For other cases - another services should be used.

Reply:
thanks, use of the services instead of  dao was implemented